File: ClassEval_T/cpp/solution/Order.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>

struct Dish {
    std::string dish;
    double price;
    int count;
};

class Order {
public:
    Order();
    bool add_dish(const Dish& dish);
    double calculate_total() const;
    double checkout();

    std::vector<Dish> menu;
    std::vector<Dish> selected_dishes;
    std::unordered_map<std::string, double> sales;
};

Order::Order() {}


bool Order::add_dish(const Dish& dish) {
    for (auto& menu_dish : menu) {
        if (dish.dish == menu_dish.dish) {
            if (menu_dish.count < dish.count) {
                return false;
            }
            else {
                menu_dish.count -= dish.count;
                break;
            }
        }
    }
    selected_dishes.push_back(dish);
    return true;
}


double Order::calculate_total() const {
    double total = 0;
    for (const auto& dish : selected_dishes) {
        auto it = sales.find(dish.dish);
        if (it != sales.end()) {
            total += dish.price * dish.count * it->second;
        }
    }
    return total;
}


double Order::checkout() {
    if (selected_dishes.empty()) {
        return 0;
    }
    double total = calculate_total();
    selected_dishes.clear();
    return total;
}

File: BatchTestTool/java/java automated tester/Test/compile_passed/solutions/Order.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Order {

    private List<Map<String, Object>> menu;
    private List<Map<String, Object>> selectedDishes;
    private Map<String, Integer> sales;
    private Map<String, Double> salesDouble;

    public Order() {
        this.menu = new ArrayList<>();
        this.selectedDishes = new ArrayList<>();
        this.sales = new HashMap<>();
        this.salesDouble = new HashMap<>();
    }

    public boolean addDish(Map<String, Object> dish) {
        for (Map<String, Object> menuDish : menu) {
            if (dish.get("dish").equals(menuDish.get("dish"))) {
                int count = (int) dish.get("count");
                if (menuDish.get("count").equals(count)) {
                    return false;
                } else {
                    menuDish.put("count", (int) menuDish.get("count") - count);
                    break;
                }
            }
        }
        selectedDishes.add(dish);
        return true;
    }

    public double calculateTotal() {
        double total = 0;
        for (Map<String, Object> dish : selectedDishes) {
            total += (double) dish.get("price") * (int) dish.get("count") * salesDouble.get((String) dish.get("dish"));
        }
        return total;
    }

    public double checkout() {
        if (selectedDishes.isEmpty()) {
            return 0;
        }
        double total = calculateTotal();
        selectedDishes.clear();
        return total;
    }
}